### PR TITLE
Add pulling of screenshots during UI Test for screenshot-tests-for-android

### PR DESCRIPTION
**Pull request by MagicCedric**
Introduce PullScreenshotTestRunListener to pull screenshots from UI-Test if the configuration has a filter that enables it. It will only execute it at the end of the batch of test, and the runner will wait until it finishes (blocking) to finish the batch.

Introduce Configuration.pullScreenshotFilterConfiguration to specify filters that mark which test contains screenshots to PullScreenshotTestRunListener.

Update version to 0.5.3

### CODE REVIEW

File: vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt [hunk 1]
```diff
@@ -301,11 +313,18 @@ class DdmlibAndroidDevice(
             prepareRecorderListener(feature, fileManager, devicePoolId, attachmentProviders)
         } ?: NoOpTestRunListener()
 
+        val vrtListener = if (configuration.isPullScreenshotEnabled()) {
```
Comment (lukaville) [inline vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt]: rename `vrtListener` to `pullScreenshotsListener`?

File: cli/src/main/kotlin/com/malinskiy/marathon/cli/config/ConfigFactory.kt [hunk 2]
```diff
@@ -67,7 +67,8 @@ class ConfigFactory(private val mapper: ObjectMapper) {
             testOutputTimeoutMillis = config.testOutputTimeoutMillis,
             debug = config.debug,
             vendorConfiguration = vendorConfiguration as VendorConfiguration,
-            analyticsTracking = config.analyticsTracking
+            analyticsTracking = config.analyticsTracking,
+            pullScreenshotFilterConfiguration = config.pullScreenshotFilterConfiguration
```
Comment (lukaville) [inline cli/src/main/kotlin/com/malinskiy/marathon/cli/config/ConfigFactory.kt:71]: we need to ignore caching for these tests as we do for strict run tests, check `TestCacheLoader:69`. We immediately emit cache miss for strict run tests. Looks like we need to do the same for screenshot tests

Comment (MagicCedric) [reply cli/src/main/kotlin/com/malinskiy/marathon/cli/config/ConfigFactory.kt:71]: True, done in a similar manner to what we do for strict run

File: vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt (hunk 1 above)
Comment (MagicCedric) [reply vendor/vendor-android/ddmlib/src/main/kotlin/com/malinskiy/marathon/android/ddmlib/DdmlibAndroidDevice.kt]: Yes, also renamed package

**Review by lukaville — APPROVED** [review]

**Pull request merged**